subsys/bluetooth/audio/has.c

```diff
@@ -868,19 +856,102 @@ static int bt_has_cp_generic_update(struct has_client *client, uint8_t prev_inde
 	}
 }
 
+#if defined(CONFIG_BT_SETTINGS)
+struct client_context_store {
+	/* Last notified preset index */
+	uint8_t last_preset_index_known;
+};
+
+static int settings_set_cb(const char *name, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	struct client_context_store store;
+	struct client_context *context;
+	bt_addr_le_t addr;
+	const char *next;
+	ssize_t len;
+	int err;
+
+	LOG_DBG("%s", name);
+
+	if (!name) {
+		LOG_ERR("Insufficient number of arguments");
+		return -EINVAL;
+	}
+
+	err = bt_settings_decode_key(name, &addr);
+	if (err) {
+		LOG_ERR("Unable to decode address %s", name);
+		return -EINVAL;
+	}
+
+	settings_name_next(name, &next);
+
+	context = context_find(&addr);
+	if (context == NULL && len_rd) {
```

Comment:
Would it make sense to check `len_rd == 0` earlier? As far as I can tell, it is the length of the settings, so if it is 0, isn't that just a no-op?

Reply:
I rearranged this part a bit, as I want to allocate the context still for all bonded devices